### ISSUE TITLE
Fix error in growth ODE test

### ISSUE THREAD

**Issue by Rosejoycrocker** [open-AIMS/ADRIA.jl#551]
Growth ode test in `../test/growth.jl` L877 errors at this test:

```julia
    @test any(diff(C_cover; dims=1) .< 0) ||
        "ODE never decreases, du being restricted to >=0."
    @test any(diff(C_cover; dims=1) .>= 0) ||
        "ODE never increases, du being restricted to <=0."

```

This is due to `p.X_mb` and `p.rec` being updated outside of the time step loop (i.e. given non-zero values) and being treated as a local variable inside the timestep loop. Due to this, inside the timestep loop `p.X_mb` and `p.rec` are matrices of zeros and hence the ODE shows no changes and the test errors. Could perhaps be fixed by moving the time step loop into a function.

**Comment by ConnectedSystems** [comment]
Test errors like these should be fixed now as it points to an issue in the model or means that the test needs to be updated due to a change. I don't think it should be done by someone new to the project with no context of how the changes relate to one another.

**Comment by Rosejoycrocker** [comment]
From what I can see it's not an issue with the ODE but due to ODE variables being out of scope when they are used in the for loop, but I'll find a way to fix it.

**Comment by Rosejoycrocker** [comment]
Closed in #573
